### ISSUE TITLE
Segmentation fault during runtime in OSX 

### ISSUE THREAD

**Issue by felipegb94** [arrayfire/arrayfire#920]
Hi,
I just installed ArrayFire through homebrew and I created the following simple program:
```
#include <stdio.h>
#include <math.h>
#include <cstdio>
#include <cstdlib>
#include <arrayfire.h>
//using namespace af;

int main(int argc, char ** argv)
{
    af::info();
   return 0;
}
```
The program compiles correctly, but when I run it gives me the following output and ends in a seg fault:
```
NSGL: The targeted version of OS X only supports forward-compatible contexts for OpenGL 3.2 and aboveError: Could not Create GLFW Window!
ArrayFire v3.0.0 (OpenCL, 64-bit Mac OSX, build )
[0] APPLE   : HD Graphics 4000 OpenCL 1.2  Device driver 1.2(Mar 27 2015 01:47:22) FP64 Support(False)
Segmentation fault: 11
```
It seems that the program is running correctly but since it is not able to create a GLFW window it crashes. I tried this in one of the examples programs (pi.cpp) and it the program ran correctly (gave the pi outputs) but ended in a seg fault.

Is this a common problem in OSX? I have OSX 10.10. 
Hopefully this is not a duplicate question (I have already been browsing the issues for a while)..

Thanks for the help!

**Comment by pavanky** [comment]
Can you try installing our binary package to see if it works: http://arrayfire.com/installer_archive/3.0.2/ArrayFire-v3.0.2_OSX.pkg

This will help us narrow some issues. Right now the topic seems too broad to narrow down.

**Comment by shehzan10** [comment]
I believe this issue was fix in 3.0.1.
If you wish to continue with the brew build of 3.0.0, you can use `AF_DISABLE_GRAPHICS=1 ./myexec`. This will disable all graphics APIs.

**Comment by felipegb94** [comment]
I installed the arrayfire pkg when I ran the same program I got the following runtime error:

```
libc++abi.dylib: terminating with uncaught exception of type AfError: CUDA Error (35): CUDA driver version is insufficient for CUDA runtime version

Abort trap: 6
```
I know you get that error when cuda is not installed or nor available, but I have an intel gpu so I need to run af with OpenCL. 

I did the `AF_DISABLE_GRAPHICS=1 ./myexec` fix and it worked (with the brew installation). I'm not going to be doing any graphics programming so I decided to go with that fix. 

Thank you very much for the help! I'm looking forward to use af in a couple of my projects :)
